src/cell.cpp

```diff
@@ -318,7 +318,12 @@ CSGCell::CSGCell(pugi::xml_node cell_node)
   // Convert user IDs to surface indices.
   for (auto& r : region_) {
     if (r < OP_UNION) {
-      r = copysign(model::surface_map[abs(r)] + 1, r);
+      const auto& it {model::surface_map.find(abs(r))};
+      if (it == model::surface_map.end()) {
+        throw std::runtime_error{"Invalid surface ID " + std::to_string(abs(r))
+          + " specified in region for cell " + std::to_string(id_) + "."};
```

Comment:
Using `throw` is a departure from our usual MO of calling `fatal_error`.  What's your reasoning on this?  (I'm not attached to `fatal_error`; I just haven't looked into it.)  If we start using `throw` elsewhere in the code do we need to take care that `abort_mpi` is called?

Reply:
It has the same effect (aborting the program) but with the benefit that if you want to catch it further up the call stack, you hypothetically could (e.g., a user of the C API). If you'd prefer we stick with `fatal_error` for now and then systematically switch them later, I'm fine with that.

Reply:
Nah, it's okay.  I'm fine with using `throw`